fix: immediately load the enhancer

kptdobe opened the pull request:
@davidnuescheler and I were looking at the RUM data of the GenStudio and only few data are available. On top of low traffic, we discovered that the enhancer is not loaded.
Reason: the `load` event has already fired.

Relying on the `load` event is problematic, because we do not know when the script `rum-standalone.js` script is loaded. In the GenStudio case, it is pretty late. 
If people follow the instructions to load the script right after the LCP, there is not specific reason to wait for the `load` event to load the enhancer. 
If they add it pre-LCP, the costly DNS lookup and SSL handshake are already done anyway (i.e. to load `rum-standalone.js`), which is another reason to not wait for the `load` event.

For all those reasons, we should get rid of the listener on the `load` event and just immediately include the enhancer.

chicharr commented:
@kptdobe this code is also used for franklin projects, not just for standalone mode. 
This change would have an impact in the performance of franklin websites.

kptdobe commented:
ok, I have changed the implementation: check if the load event already occurred. I have also added the corresponding tests.

I suggest we go with this approach as quick fix but we define a more reliable and stable approach and change the implementation in a near future.
